applications/FluidDynamicsApplication/custom_elements/two_fluid_navier_stokes.cpp

```diff
@@ -0,0 +1,2054 @@
+const double crhs3 =             DN(0,0)*v(0,0) + DN(1,0)*v(1,0) + DN(2,0)*v(2,0);
+const double crhs4 =             N[0]*vconv(0,0) + N[1]*vconv(1,0) + N[2]*vconv(2,0);
+const double crhs5 =             N[0]*vconv(0,1) + N[1]*vconv(1,1) + N[2]*vconv(2,1);
+const double crhs6 =             rho*(crhs3*crhs4 + crhs5*(DN(0,1)*v(0,0) + DN(1,1)*v(1,0) + DN(2,1)*v(2,0)));
+const double crhs7 =             DN(0,1)*v(0,1) + DN(1,1)*v(1,1) + DN(2,1)*v(2,1);
+const double crhs8 =             crhs3 + crhs7;
+const double crhs9 =             rho*stab_c2*sqrt(pow(crhs4, 2) + pow(crhs5, 2));
+const double crhs10 =             crhs8*(crhs9*h/stab_c1 + mu);
+const double crhs11 =             rho*(DN(0,0)*crhs4 + DN(0,1)*crhs5);
+const double crhs12 =             1.0/(crhs9/h + mu*stab_c1/pow(h, 2) + dyn_tau*rho/dt);
+const double crhs13 =             1.0*crhs12*(DN(0,0)*p[0] + DN(1,0)*p[1] + DN(2,0)*p[2] - crhs1 + crhs2 + crhs6);
+const double crhs14 =             rho*(N[0]*f(0,1) + N[1]*f(1,1) + N[2]*f(2,1));
+const double crhs15 =             rho*(N[0]*(bdf0*v(0,1) + bdf1*vn(0,1) + bdf2*vnn(0,1)) + N[1]*(bdf0*v(1,1) + bdf1*vn(1,1) + bdf2*vnn(1,1)) + N[2]*(bdf0*v(2,1) + bdf1*vn(2,1) + bdf2*vnn(2,1)));
+const double crhs16 =             rho*(crhs4*(DN(0,0)*v(0,1) + DN(1,0)*v(1,1) + DN(2,0)*v(2,1)) + crhs5*crhs7);
+const double crhs17 =             1.0*crhs12*(DN(0,1)*p[0] + DN(1,1)*p[1] + DN(2,1)*p[2] - crhs14 + crhs15 + crhs16);
+const double crhs18 =             rho*(DN(1,0)*crhs4 + DN(1,1)*crhs5);
+const double crhs19 =             rho*(DN(2,0)*crhs4 + DN(2,1)*crhs5);
+            rhs[0]=DN(0,0)*crhs0 - DN(0,0)*crhs10 - DN(0,0)*stress[0] - DN(0,1)*stress[2] + N[0]*crhs1 - N[0]*crhs2 - N[0]*crhs6 - crhs11*crhs13;
+            rhs[1]=-DN(0,0)*stress[2] + DN(0,1)*crhs0 - DN(0,1)*crhs10 - DN(0,1)*stress[1] + N[0]*crhs14 - N[0]*crhs15 - N[0]*crhs16 - crhs11*crhs17;
+            rhs[2]=-DN(0,0)*crhs13 - DN(0,1)*crhs17 - N[0]*crhs8;
+            rhs[3]=DN(1,0)*crhs0 - DN(1,0)*crhs10 - DN(1,0)*stress[0] - DN(1,1)*stress[2] + N[1]*crhs1 - N[1]*crhs2 - N[1]*crhs6 - crhs13*crhs18;
+            rhs[4]=-DN(1,0)*stress[2] + DN(1,1)*crhs0 - DN(1,1)*crhs10 - DN(1,1)*stress[1] + N[1]*crhs14 - N[1]*crhs15 - N[1]*crhs16 - crhs17*crhs18;
+            rhs[5]=-DN(1,0)*crhs13 - DN(1,1)*crhs17 - N[1]*crhs8;
+            rhs[6]=DN(2,0)*crhs0 - DN(2,0)*crhs10 - DN(2,0)*stress[0] - DN(2,1)*stress[2] + N[2]*crhs1 - N[2]*crhs2 - N[2]*crhs6 - crhs13*crhs19;
+            rhs[7]=-DN(2,0)*stress[2] + DN(2,1)*crhs0 - DN(2,1)*crhs10 - DN(2,1)*stress[1] + N[2]*crhs14 - N[2]*crhs15 - N[2]*crhs16 - crhs17*crhs19;
+            rhs[8]=-DN(2,0)*crhs13 - DN(2,1)*crhs17 - N[2]*crhs8;
+
+
+    noalias(rRHS) += rData.Weight * rhs;
+
+}
+
+
+template <>
+void TwoFluidNavierStokes<TwoFluidNavierStokesData<3, 4>>::ComputeGaussPointRHSContribution(
+	TwoFluidNavierStokesData<3, 4>& rData, VectorType& rRHS) {
+
+    const double rho = rData.Density;
+    const double mu = rData.EffectiveViscosity;

```

Comment:
This is the criteria that was stablished in the old two fluid and the stokes element. In practice, changing this would make this element unusable for c2c.

Reply:
I guessed that... If this is the case, we need to bear in mind to switch the distance sign any time we use the core distance algorithms.